apm-collector/apm-collector-ui/collector-ui-jetty-provider/src/main/java/org/apache/skywalking/apm/collector/ui/service/ServiceTopologyService.java

```diff
@@ -86,7 +86,11 @@ public Topology getServiceTopology(Step step, int serviceId, long startTimeBucke
             Call call = new Call();
             call.setSource(referenceMetric.getSource());
             call.setTarget(referenceMetric.getTarget());
-            call.setAvgResponseTime((referenceMetric.getDurations() - referenceMetric.getErrorDurations()) / (referenceMetric.getCalls() - referenceMetric.getErrorCalls()));
+            if (referenceMetric.getCalls() == referenceMetric.getErrorCalls()) {
+                call.setAvgResponseTime(referenceMetric.getDurations() / referenceMetric.getCalls());
```

Comment:
This process way has conflicts with other. If all requests are error, the avg response time should be 0.

Reply:
fixed.avg time has been set to 0